src/main/kotlin/gg/essential/elementa/components/UIText.kt

```diff
@@ -15,14 +16,16 @@ import java.awt.Color
  * Simple text component that draws its given `text` at the scale determined by
  * this component's width & height constraints.
  */
-open class UIText @JvmOverloads constructor(
-    text: String = "",
-    shadow: Boolean = true,
-    shadowColor: Color? = null
-) : UIComponent() {
-    private val textState = BasicState(text).map { it } // extra map so we can easily rebind it
-    private var shadowState: State<Boolean> = BasicState(shadow)
-    private var shadowColorState: State<Color?> = BasicState(shadowColor)
+open class UIText constructor(text: State<String>, shadow: State<Boolean>, shadowColor: State<Color?>) : UIComponent() {
```

Comment:
I'd seem useful to provide defaults for these like the other constructor does (except for the first one, otherwise the no-arg call will be ambiguous).